### DISCUSSION TITLE
How to fetch offers through Seaport?

### DISCUSSION THREAD

**Discussion by ochhii1337** [ProjectOpenSea/seaport#881] [Q&A] [answered]
Hi frens, 

I am creating a smart contract that will call function `fulfillBasicOrder` of Seaport contract and I have limited fields in my smart contract's function, only contractAddress and the tokenId.

I believe that listings are stored on-chain so how can I retrieve the NFT price without using Opensea API? 

Other required fields include additional recipients. Can these fields be retrieved on-chain too?

Many thanks,
Ochhii

**Comment by ryanio** [comment, marked as answer]
Listings are not on chain, except for on chain validated orders, to fulfill off chain listings use https://docs.opensea.io/reference/fulfill-a-listing
